compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml

```diff
@@ -215,20 +215,23 @@ instance Action Scenario where
 instance ActionFail Scenario where
     fail = primitive @"SAbort"
 
-class HasSubmit m cmds | m -> cmds, cmds -> m where
+class HasSubmit m cmds discs | m -> cmds, m -> discs, cmds -> m where
   -- | `submit p cmds` submits the commands `cmds` as a single transaction
   -- from party `p` and returns the value returned by `cmds`.
   --
   -- If the transaction fails, `submit` also fails.
   submit : HasCallStack => Party -> cmds a -> m a
 
+  submitDisclosing : HasCallStack => discs -> Party -> cmds a -> m a
+  submitDisclosing _ds p cmds = submit p cmds
```

Comment:
that default implementation is broken since it doesn’t actually use the disclosed contracts. I think if we want to add this to this typeclass, we should error out on scenarios. However, I don’t see why we need to do this at all. This could just be a top-level definition in `Daml.Script`.